cli/azd/pkg/azsdk/storage/storage_blob_client.go

```diff
@@ -0,0 +1,208 @@
+	client *azblob.Client
+}
+
+// Blob represents a blob within a storage account container.
+type Blob struct {
+	Name         string
+	Path         string
+	CreationTime time.Time
+	LastModified time.Time
+}
+
+// Items returns a list of blobs in the configured storage account container.
+func (bc *blobClient) Items(ctx context.Context) ([]*Blob, error) {
+	blobs := []*Blob{}
+
+	pager := bc.client.NewListBlobsFlatPager(bc.config.ContainerName, nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			err = fmt.Errorf("failed to get next page of blobs, %w", err)
+			var responseErr *azcore.ResponseError
+			if errors.As(err, &responseErr) {
+				return nil, fmt.Errorf("%w, %w", err, ErrContainerNotFound)
+			}
+			return nil, err
+		}
+
+		for _, blob := range page.Segment.BlobItems {
+			blobs = append(blobs, &Blob{
+				Name:         filepath.Base(*blob.Name),
+				Path:         *blob.Name,
+				CreationTime: *blob.Properties.CreationTime,
+				LastModified: *blob.Properties.LastModified,
+			})
+		}
+	}
+
+	return blobs, nil
+}
+
+// Download downloads a blob from the configured storage account container.
+func (bc *blobClient) Download(ctx context.Context, blobPath string) (io.ReadCloser, error) {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return nil, err
+	}
+
+	resp, err := bc.client.DownloadStream(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download blob '%s', %w", blobPath, err)
+	}
+
+	return resp.Body, nil
+}
+
+// Upload uploads a blob to the configured storage account container.
+func (bc *blobClient) Upload(ctx context.Context, blobPath string, reader io.Reader) error {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err := bc.client.UploadStream(ctx, bc.config.ContainerName, blobPath, reader, nil)
+	if err != nil {
+		return fmt.Errorf("failed to upload blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Delete deletes a blob from the configured storage account container.
+func (bc *blobClient) Delete(ctx context.Context, blobPath string) error {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err := bc.client.DeleteBlob(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Check if the specified container exists
+// If it doesn't already exist then create it
+func (bc *blobClient) ensureContainerExists(ctx context.Context) error {
+	exists := false
+
+	pager := bc.client.NewListContainersPager(nil)
+	for pager.More() {
+		if exists {
+			break
+		}
+
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed getting next page of containers: %w", err)
+		}
+
+		for _, container := range page.ContainerItems {
+			if *container.Name == bc.config.ContainerName {
+				exists = true
+				break
+			}
+		}
+	}
+
+	if !exists {
```

Comment:
Some of the other language SDKs provide a CreateIfNotExists method which works by trying to create the container and then ignoring the ContainerAlreadyExists error that comes back: 

https://github.com/Azure/azure-sdk-for-net/blob/d4e55222514a6c310db25fd2df0415cd0aacf708/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L910-L1119

Perhaps that approach is better here?  The above does have a TOCTOU bug, since the container could be created (or I guess removed) between when we start the page operation and when we complete it.

Might be reasonable to just follow the SDK Team's lead here.

Reply:
The Go sdk doesn't include a `CreateIfNotExists` error.  I tried just ignoring the case where the container already exists but this slows things down because it forces the internal retry strategy to retry multiple times causing a more significant delay.

Going to keep as is for the moment.